mjcf_to_sdformat/mjcf_to_sdformat/converters/geometry.py

```diff
@@ -0,0 +1,130 @@
+    elif geom.type == "capsule":
+        capsule = sdf.Capsule()
+        capsule.set_radius(geom.size[0])
+        if geom.fromto is None:
+            capsule.set_length(geom.size[1] * 2)
+        else:
+            v1 = Vector3d(geom.fromto[0], geom.fromto[1], geom.fromto[2])
+            v2 = Vector3d(geom.fromto[3], geom.fromto[4], geom.fromto[5])
+            length = v1.distance(v2)
+            capsule.set_length(length)
+        sdf_geometry.set_capsule_shape(capsule)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.CAPSULE)
+    elif geom.type == "cylinder":
+        cylinder = sdf.Cylinder()
+        cylinder.set_radius(geom.size[0])
+        if geom.fromto is None:
+            cylinder.set_length(geom.size[1] * 2)
+        else:
+            v1 = Vector3d(geom.fromto[0], geom.fromto[1], geom.fromto[2])
+            v2 = Vector3d(geom.fromto[3], geom.fromto[4], geom.fromto[5])
+            length = v1.distance(v2)
+            cylinder.set_length(length)
+        sdf_geometry.set_cylinder_shape(cylinder)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.CYLINDER)
+    elif geom.type == "ellipsoid":
+        ellipsoid = sdf.Ellipsoid()
+        ellipsoid.set_radii(su.list_to_vec3d(geom.size))
+        sdf_geometry.set_ellipsoid_shape(ellipsoid)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.ELLIPSOID)
+    elif geom.type == "sphere":
+        sphere = sdf.Sphere()
+        sphere.set_radius(geom.size[0])
+        sdf_geometry.set_sphere_shape(sphere)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.SPHERE)
+    elif geom.type == "plane":
+        plane = sdf.Plane()
+        plane.set_size(Vector2d(geom.size[0] * 2, geom.size[1] * 2))
+        sdf_geometry.set_plane_shape(plane)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.PLANE)
+    else:
+        raise RuntimeError(
+            f"Encountered unsupported shape type {geom.type}")
+    return sdf_geometry
+
+
+def add_mjcf_visual_to_sdf(geom):
+    """
+    Converts MJCF geom to a SDFormat visual
+    MJCF geom should be part of group `VISUAL_GEOM_GROUP`.
+
+    :param mjcf.Element geom: The MJCF geom.
+    :return: The newly created SDFormat visual.
+    :rtype: sdformat.Visual
+    """
+    visual = sdf.Visual()
+    if geom.name is not None:
+        visual.set_name("visual_" + geom.name)
+    else:
+        global VISUAL_NUMBER
+        visual.set_name("visual_" + str(VISUAL_NUMBER))
+        VISUAL_NUMBER = VISUAL_NUMBER + 1
+    sdf_geometry = add_mjcf_geometry_to_sdf(geom)
+    if sdf_geometry is not None:
+        visual.set_geometry(sdf_geometry)
+    else:
+        return None
+    return visual
+
+
+def add_mjcf_collision_to_sdf(geom):
+    """
+    Converts MJCF geom to a SDFormat collision
+    MJCF geom should be part of group `COLLISION_GEOM_GROUP`.
+
+    :param mjcf.Element geom: The MJCF geom.
+    :return: The newly created SDFormat collision.
+    :rtype: sdformat.Collision
+    """
+    col = sdf.Collision()
```

Comment:
Should the pose of the collision be set here from the geom?

Reply:
https://github.com/gazebosim/gz-mujoco/blob/40148462713bdbba5fa3ba236d57bf8c24596e61/mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py#L99-L128